x/wasm/ibc.go

```diff
@@ -267,23 +268,38 @@ func (i IBCHandler) OnRecvPacket(
 		return channeltypes.NewErrorAcknowledgement(errorsmod.Wrapf(err, "contract port id"))
 	}
 	msg := wasmvmtypes.IBCPacketReceiveMsg{Packet: newIBCPacket(packet), Relayer: relayer.String()}
-	ack, err := i.keeper.OnRecvPacket(ctx, contractAddr, msg)
+
+	em := sdk.NewEventManager()
+	ack, err := i.keeper.OnRecvPacket(ctx.WithEventManager(em), contractAddr, msg)
 	if err != nil {
+		// the state gets reverted, we keep only wasm events that do not contain custom data
+		for _, e := range em.Events() {
+			if types.IsAcceptedEventOnRecvPacketErrorAck(e.Type) {
+				ctx.EventManager().EmitEvent(e)
+			}
+		}
```

Comment:
Do we really want to preserve those events? I might be messing something but to me it's clearer if we discard them and add all relevent information in the error event below.

Reply:
In a more complex contract-calls-contracts environment, the event trace may have revealed some infos to track the root cause of an error. But there are some costs of maintaining this in the future. 
👌 let's drop them and keep things simple.